OnPageCode.js

```diff
@@ -1,196 +1,334 @@
+  var PM_Network_POC = {
+
+    // IMP: Which namespace to be used, can be identify using,
+    // 1. Go to publisher URL.
+    // 2. Open developer tool and select console tab
+    // 3. Execute below code in the console,
+    //    3.1 window.owpbjs
+    //    If it returns object with some properties, use owpbjs as prebidNamespace
+    //    else, try changing namespace with pbjs (code: window.pbjs)
+    //    3.2 In case of IDHUB only, use code, window.ihowpbjs
+    // 4. Or connect to JS developer
+
+    'prebidNamespace': 'pbjs',
+
+    'publisherId': 1234,
+
+    // N second after auction end, get the stats for given domains
+    'executionDelayInMs': 2000,
+
+    // run the Network analysis only for the mentioned percentage of traffic
+    'testGroupPercentage': 100,
+
+    'statHatParameters': [
+      {
+        key: 'dns',
+        name: 'DNS Lookup',
+        timeEndKey: 'domainLookupEnd',
+        timeStartKey: 'domainLookupStart'
+      },
+      {
+        key: 'tcp',
+        name: 'TCP Connection',
+        timeEndKey: 'connectEnd',
+        timeStartKey: 'connectStart'
+      },
+      {
+        key: 'que_st',
+        name: 'Queueing and stalled',
+        timeEndKey: 'requestStart',
+        timeStartKey: 'startTime'
+      },
+      {
+        key: 'rs_wfs',
+        name: 'Request Sent and Waiting For Server',
+        timeEndKey: 'responseStart',
+        timeStartKey: 'requestStart'
+      },
+      {
+        key: 'cd',
+        name: 'Content Download',
+        timeEndKey: 'responseEnd',
+        timeStartKey: 'responseStart'
+      },
+      {
+        key: 'dur',
+        name: 'Duration',
+        timeEndKey: 'responseEnd',
+        timeStartKey: 'startTime'
+      },
+      {
+        key: 'per_lt',
+        name: 'Perceived Latency',
+        timeEndKey: 'responseEnd',
+        timeStartKey: 'fetchStart'
+      }
+    ],
+
+    'bidders': [],
+
+    'bidderTracker': {},
+
+    'order': 0,
+
+    'adUnitCount': 0,
+
+    // need to log domain in stats
+    'domain': (function () {
+      const replaceList = ['https://', 'http://'];
+      let hostName = window?.location?.hostname;
+      replaceList.forEach(replaceThis => {
+        hostName = hostName.replace(replaceThis, '');
+      });
+      return hostName;
+    })(),
+
+    'browserName': (function () {
+      let userAgent = navigator?.userAgent || 'others';
+      if (userAgent.match(/firefox|fxios/i)) return 'firefox';
+      if (userAgent.match(/chrome|chromium|crios/i)) return 'chrome';
+      if (userAgent.match(/safari/i)) return 'safari';
+      if (userAgent.match(/opr\//i)) return 'opera';
+      if (userAgent.match(/edg/i)) return 'edge';
+      return 'others';
+    })(),
+
+    'platform': (function () {
+      return getDeviceType();
+    })(),
+
+    // to make sure that we do not read the stats for the same network call again
+    'lastExecutionMaxIndex': 0,
+
+    'networkType': navigator.connection.effectiveType,
```

Comment:
are we sure that the `navigator.connection.effectiveType,` is supported by all browsers? I feel we should handle the case where it is not supported by a browser.

Reply:
good call to play it safe on this.  i can look into compatibility around `navigator.connection.effectiveType`